e2e/cypress/integration/plugins/marketplace/marketplace_spec.js

```diff
@@ -300,4 +330,47 @@ describe('Plugin Marketplace', () => {
             cy.get('#marketplace-plugin-github').should('be.visible');
         });
     });
+
+    describe('EnableRemoteMarketplace disabled, should', () => {
+        beforeEach(() => {
+            const newSettings = {
+                PluginSettings: {
+                    Enable: true,
+                    EnableMarketplace: true,
+                    EnableRemoteMarketplace: false,
+                    MarketplaceUrl: 'https://api.integrations.mattermost.com',
+                },
+            };
+            cy.apiUpdateConfig(newSettings);
+
+            // # Login as sysadmin
+            cy.apiLogin('sysadmin');
+            cy.visit('/');
+
+            // # Click hamburger main menu
+            cy.get('#sidebarHeaderDropdownButton').click();
+
+            // # Open up marketplace modal
+            cy.get('#marketplaceModal').click();
+        });
+
+        it('not display any plugins and no error bar', () => {
+            // * no plugins should be visible
+            cy.get('#marketplaceTabs-pane-allPlugins').find('.more-modal__row').should('have.length', 0);
```

Comment:
Initial render shows loading indicator which will immediately assess this test as pass.  I suggest to add another selector to ensure that the modal list is rendered before checking no row on it.
```suggestion
            cy.get('#marketplaceTabs-pane-allPlugins > .more-modal__list').find('.more-modal__row').should('have.length', 0);
```

or better if you have text indicator that's visible to user.
```suggestion
            cy.get('#marketplaceTabs-pane-allPlugins').findByText('There are no plugins available at this time.');
```